packages/amplify_datastore/lib/method_channel_datastore.dart

```diff
@@ -23,18 +23,22 @@ const MethodChannel _channel = MethodChannel('com.amazonaws.amplify/datastore');
 
 /// An implementation of [AmplifyDataStore] that uses method channels.
 class AmplifyDataStoreMethodChannel extends AmplifyDataStore {
-  var _allModelsStreamFromMethodChannel = null;
+  dynamic _allModelsStreamFromMethodChannel = null;
+
+  /// Internal use constructor
+  AmplifyDataStoreMethodChannel() : super.protected();
 
   /// This method adds model schemas which is necessary to instantiate native plugins
   /// This is needed before the Amplify.configure() can be called, since the native
   /// plugins are needed to be added before that.
   Future<void> configureModelProvider(
-      {ModelProviderInterface modelProvider}) async {
+      {ModelProviderInterface? modelProvider}) async {
     try {
       return await _channel
           .invokeMethod('configureModelProvider', <String, dynamic>{
-        'modelSchemas':
-            modelProvider.modelSchemas.map((schema) => schema.toMap()).toList(),
+        'modelSchemas': modelProvider!.modelSchemas
+            .map((schema) => schema.toMap())
+            .toList(),
         'modelProviderVersion': modelProvider.version

```

Comment:
Weird that this didn't need a cast or null-safe access ...

Reply:
Yeah it seems that if you force cast previously the compiler doesn't both enforcing it for following accesses...